### PR TITLE
shared: create inaccessible files with correct security label

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: src/shared/label-util.c [hunk 1]
```diff
@@ -99,6 +99,24 @@ int mknod_label(const char *pathname, mode_t mode, dev_t dev) {
         return mac_smack_fix(pathname, 0);
 }
 
+int mknodat_label(int dirfd, const char *pathname, mode_t mode, dev_t dev) {
```
Comment (YHNdnzj) [inline src/shared/label-util.c:84]: Please make mknod_label a static inline wrapper of this function

File: src/shared/label-util.c [hunk 2]
```diff
@@ -99,6 +99,24 @@ int mknod_label(const char *pathname, mode_t mode, dev_t dev) {
...
+        int r;
+
+        assert(pathname);
+
+        r = mac_selinux_create_file_prepare_at(dirfd, pathname, mode);
+        if (r < 0)
+                return r;
+
+        r = RET_NERRNO(mknodat(dirfd, pathname, mode, dev));
```
Comment (YHNdnzj) [inline src/shared/label-util.c:94]: I.e., please `assert(dirfd >= 0 || dirfd == AT_FDCWD)` above, and pass AT_FDCWD for mknod_label

File: src/basic/fs-util.h [hunk 3]
```diff
@@ -128,15 +128,21 @@ int posix_fallocate_loop(int fd, uint64_t offset, uint64_t size);
...
         XO_SUBVOLUME = 1 << 1,
 } XOpenFlags;
 
+int open_mkdir_at_full(int dirfd, const char *path, int flags, XOpenFlags xopen_flags, mode_t mode);
+static inline int open_mkdir_at(int dirfd, const char *path, int flags, mode_t mode) {
+        return open_mkdir_at_full(dirfd, path, flags, 0, mode);
+}
+static inline int open_mkdir_at_label(int dirfd, const char *path, int flags, mode_t mode) {
+        return open_mkdir_at_full(dirfd, path, flags, XO_LABEL, mode);
+}
```
Comment (poettering) [inline src/basic/fs-util.h]: honestly if we just have a single caller of this maybe just spell things out in the call site, i.e. call `open_mkdir_at_full(…, XO_LABEL, …)` directly there and avoid this static inlne wrpper?

File: src/shared/label-util.c (hunk 1 above)
Comment (cgzones) [reply src/shared/label-util.c:84]: done

**Pull request merged**